pandas/core/reshape/merge.py

```diff
@@ -689,7 +689,7 @@ def get_result(self):
 
         self._maybe_restore_index_levels(result)
 
-        return result
+        return result.__finalize__(self, method="merge")
```

Comment:
`self` here is a `_MergeOperation` so the usual `__finalize__` won't do anything with hit. Maybe leave this one off for now.